cargo-dist/src/backend/installer/homebrew.rs

```diff
@@ -265,4 +326,99 @@ mod tests {
     fn ampersand_but_no_digit() {
         run_comparison("openssl@blah", "Openssl@blah");
     }
+
+    fn run_spdx_comparison(spdx_string: &str, homebrew_dsl: &str) {
+        let result = to_homebrew_license_format(spdx_string).unwrap();
+        assert_eq!(result, homebrew_dsl);
+    }
+
+    #[test]
+    fn spdx_single_license() {
+        run_spdx_comparison("MIT", r#""MIT""#);
+    }
+
+    #[test]
+    fn spdx_single_license_with_plus() {
+        run_spdx_comparison("Apache-2.0+", r#""Apache-2.0+""#);
+    }
+
+    #[test]
+    fn spdx_two_licenses_any() {
+        run_spdx_comparison("MIT OR 0BSD", r#"any_of: ["MIT", "0BSD"]"#);
+    }
+
+    #[test]
+    fn spdx_two_licenses_all() {
+        run_spdx_comparison("MIT AND 0BSD", r#"all_of: ["MIT", "0BSD"]"#);
+    }
+
+    #[test]
+    fn spdx_two_licenses_with_plus() {
+        run_spdx_comparison("MIT OR EPL-1.0+", r#"any_of: ["MIT", "EPL-1.0+"]"#);
+    }
+
+    #[test]
+    fn spdx_three_licenses() {
+        run_spdx_comparison(
+            "MIT OR Apache-2.0 OR CC-BY-4.0",
+            r#"any_of: ["MIT", "Apache-2.0", "CC-BY-4.0"]"#,
+        );
+    }
+
+    #[test]
+    fn spdx_three_licenses_or_and() {
+        run_spdx_comparison(
+            "MIT OR Apache-2.0 AND CC-BY-4.0",
+            r#"any_of: ["MIT", { all_of: ["Apache-2.0", "CC-BY-4.0"] }]"#,
+        );
+    }
+
+    #[test]
+    fn spdx_three_licenses_and_or() {
+        run_spdx_comparison(
+            "MIT AND Apache-2.0 OR CC-BY-4.0",
+            r#"any_of: [{ all_of: ["MIT", "Apache-2.0"] }, "CC-BY-4.0"]"#,
```

Comment:
Likewise, Homebrew parses this to

```ruby
{:all_of=>["MIT", {:any_of=>["Apache-2.0", "CC-BY-4.0"]}]}
```

Reply:
Actually scratch what I said about left-to-right, this case would indicate that they _are_ looking at operator precedence, but just have OR/AND swapped.